index.js

```diff
@@ -2995,66 +3135,241 @@ function Env(handle) {
     };
 })();
 
-let _artMethodSpec = null;
-function getArtMethodSpec(vm) {
-    if (_artMethodSpec !== null)
-        return _artMethodSpec;
-
-    vm.perform(() => {
-        const env = vm.getEnv();
-        const process = env.findClass("android/os/Process");
-        const setArgV0 = env.getStaticMethodId(process, "setArgV0", "(Ljava/lang/String;)V");
+function resolveStructSpec(vm) {
+    if (this._current !== undefined)
+        return this._current;
 
-        const runtimeModule = Process.getModuleByName("libandroid_runtime.so");
-        const runtimeStart = runtimeModule.base;
-        const runtimeEnd = runtimeStart.add(runtimeModule.size);
+    let specByPointerSize = this[runtime.androidVersion.substr(0, 4)];
+    if (specByPointerSize === undefined) {
+        if (this._default) {
+            let spec = this._default(vm);
+            if (k) {
+                this._current = spec;
+                return spec;
+            }
+        }
+        specByPointerSize = this.git;
+    }
 
-        const expectedAccessFlags = kAccPublic | kAccStatic | kAccFinal | kAccNative;
+    this._current = specByPointerSize && specByPointerSize[Process.pointerSize];
+    return this._current;
+}
 
-        let jniCodeOffset = -1;
-        let accessFlagsOffset = -1;
-        let remaining = 2;
-        for (let offset = 0; offset !== 64 && remaining !== 0; offset += 4) {
-            const field = setArgV0.add(offset);
+const getArtMethodSpec = resolveStructSpec.bind({
```

Comment:
Ahh gotcha. That sounds good. I have a 64-bit device running 6.0, so I can have a look at figuring out those offsets. But first, let's merge this. Awesome work!